htdocs/adherents/stats/geo.php

```diff
@@ -74,12 +74,12 @@
         $tab = 'statscountry';
 
         $data = array();
-        $sql .= "SELECT COUNT(d.rowid) as nb, MAX(d.datevalid) as lastdate, MAX(s.dateadh) as lastsubscriptiondate, c.code, c.label";
+        $sql .= "SELECT COUNT(DISTINCT d.rowid) as nb, MAX(d.datevalid) as lastdate, MAX(s.dateadh) as lastsubscriptiondate, c.code, c.label";
         $sql .= " FROM ".MAIN_DB_PREFIX."adherent as d";
         $sql .= " LEFT JOIN ".MAIN_DB_PREFIX."c_country as c on d.country = c.rowid";
         $sql .= " LEFT JOIN ".MAIN_DB_PREFIX."subscription as s ON s.fk_adherent = d.rowid";
         $sql .= " WHERE d.entity IN (".getEntity('adherent').")";
-        $sql .= " AND d.statut = 1";
+        $sql .= " AND CURDATE() BETWEEN s.dateadh AND s.datef";
```

Comment:
what if s.datef is not defined ?
for example when no subscription was yet recorded. This seems to exclude such record.
Also CURDATE is not allowed since it uses the timezone of database that differs on timezone of application. Try to use php date dol_now().

Reply:
Un my opinion, if no subscription is recorded, then the man is a prospect, not a member not yet an adherent